middleware/kubernetes/kubernetes.go

```diff
@@ -190,11 +190,27 @@ func (k *Kubernetes) getZoneForName(name string) (string, []string) {
 // stripSRVPrefix separates out the port and protocol segments, if present
 // If not present, assume all ports/protocols (e.g. wildcard)
 func stripSRVPrefix(name []string) (string, string, []string) {
-	if name[0][0] == '_' && name[1][0] == '_' {
-		return name[0][1:], name[1][1:], name[2:]
+
+	var srv, prt string
+
+	if len(name) != 5 {
```

Comment:
Are you stripping off the {zone} before doing these counts? While `cluster.local` is the usual one we use, people can use whatever they want - it could be `east.us.cloud.enterprise.com` or whatever.